src/python/pants/backend/python/targets/python_distribution.py

```diff
@@ -0,0 +1,66 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+
+from twitter.common.collections import maybe_list
+
+from pants.base.exceptions import TargetDefinitionException
+from pants.base.payload import Payload
+from pants.base.payload_field import PrimitiveField
+from pants.build_graph.target import Target
+
+class PythonDistribution(Target):
+  """A Python distribution target that accepts a user-defined setup.py."""
+
+  default_sources_globs = '*.py'
+
+  @classmethod
+  def alias(cls):
+    return 'python_dist'
+
+  def __init__(self,
+               address=None,
+               payload=None,
+               sources=None,
+               compatibility=None,
+               **kwargs):
+    """
+    :param dependencies: The addresses of targets that this target depends on.
```

Comment:
> dependencies, if permitted by Target.__init__(), should also be disallowed for this target type for now I think.

Can you elaborate? Do you think dependencies for `PythonDistribution` shouldn't be allowed, period? If so, why? I think for our internal use case, they'd like to pass in a 3rd party dep, so leaving it out could break that use case. Or maybe they could work around it by passing it into a library/binary that depends on `PythonDistribution`, not sure.

Reply:
to add a small detail, the python code for Cortex's use case depends on tensorflow, which needs to be either a dep in `python_dist` or a dep in the consuming client. It seems like at the very least we should allow `3rdparty/...` in dependencies field of python_dist and block non-3rdparty targets?

Reply:
I removed dependencies from the docstring. I assume `PythonTarget`'s docstring should get updated too since it doesn't align? https://github.com/pantsbuild/pants/blob/master/src/python/pants/backend/python/targets/python_target.py#L71

Question about dependencies from above still relevant. I think `Target` allows you to pass in deps via `kwargs`.